dev/com.ibm.ws.kernel.service/test/com/ibm/wsspi/kernel/service/utils/MetaTypeUtilsTest.java

```diff
@@ -384,4 +401,289 @@ public void testLocale() throws Exception {
+
+        //Test Serializable
+        testSchedule = MetatypeUtils.evaluateSchedule("MON-FRI 8:00-17:00 America/Chicago")[0];
+        try {
+            //Write testSchedule to ByteArray
+            ByteArrayOutputStream bf = new ByteArrayOutputStream();
+            ObjectOutputStream oos = new ObjectOutputStream(bf);
+            oos.writeObject(testSchedule);
+            oos.close();
+
+            //Read object fromByteArray
+            ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(bf.toByteArray()));
+            expectedSchedule = (SerializableSchedule) ois.readObject();
+            assertEquals(expectedSchedule, testSchedule);
+        } catch (Exception e) {
+            fail("Failed to serialize or deserialze SerializableSchedule");
+        }
+    }
+
+    @Test
+    public void testScheduleSanitation() throws Exception {
+        SerializableSchedule testSchedule;
+        SerializableSchedule expectedSchedule;
+
+        try {
+            testSchedule = MetatypeUtils.evaluateSchedule("  mon  8:00  ")[0];
+            expectedSchedule = new SerializableSchedule(DayOfWeek.MONDAY, null, LocalTime.of(8, 0), null, null);
+            assertEquals(testSchedule, expectedSchedule);
+        } catch (Exception e) {
+            fail("Leading and trailing space should be sanitized: " + e.getMessage());
+        }
+
+        try {
+            testSchedule = MetatypeUtils.evaluateSchedule("mon   8:00")[0];
+            expectedSchedule = new SerializableSchedule(DayOfWeek.MONDAY, null, LocalTime.of(8, 0), null, null);
+            assertEquals(testSchedule, expectedSchedule);
+        } catch (Exception e) {
+            fail("Extra interal space should be sanitized: " + e.getMessage());
+        }
+
+        try {
+            testSchedule = MetatypeUtils.evaluateSchedule("mon - fri  8:00 - 10:00")[0];
+            expectedSchedule = new SerializableSchedule(DayOfWeek.MONDAY, DayOfWeek.FRIDAY, LocalTime.of(8, 0), LocalTime.of(10, 0), null);
+            assertEquals(testSchedule, expectedSchedule);
+        } catch (Exception e) {
+            fail("Spaces surrounding hyphens should be sanitized: " + e.getMessage());
+        }
+
+        try {
+            testSchedule = MetatypeUtils.evaluateSchedule("  mon  -  fri    8:00   -   10:00   GMT-6  ")[0];
+            expectedSchedule = new SerializableSchedule(DayOfWeek.MONDAY, DayOfWeek.FRIDAY, LocalTime.of(8, 0), LocalTime.of(10, 0), ZoneId.of("GMT-6"));
+            assertEquals(testSchedule, expectedSchedule);
+        } catch (Exception e) {
+            fail("All extra space should be sanitized: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void testScheduleExceptions() throws Exception {
+
+        //Constructor exceptions
+        try {
+            //At minimum need startDay and startTime
+            new SerializableSchedule(null, null, null, null, null);
+        } catch (Exception e) {
```

Comment:
A general comment for the tests in this method is that they are missing enforcement that an exception is raised.